src/DSharp.Compiler/CodeModel/ParseNodeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace DSharp.Compiler.CodeModel
+{
+    internal static class ParseNodeExtensions
+    {
+        public static T As<T>(this ParseNode self)
+            where T : ParseNode
+        {
+            return (T)self;
```

Comment:
Maybe do a as T to prevent casting exceptions and make the consumers null check it?